### PR TITLE
Added Flavors to docs

### PR DESCRIPTION
**Pull request by v-jizhang**
Added Flavors to docs

### CODE REVIEW

**Comment by v-jizhang** [comment]
Related to issue #1140 and #1526.

File: docs/concept/flavors.soy [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+<p>
+Flavors in Buck are a way to specify different variations of a build that 
+otherwise share most configuration specifications.
+</p>
+
+<p>
+Flavors can also be used to simultaneously build for two different platforms, 
+such as iOS and watchOS. Being able to build for both of these at the same 
+time is important because our iOS application has a dependency on the Watch
```
Comment (zpao) [inline docs/concept/flavors.soy:39]: "our" doesn't make sense for public docs

File: docs/concept/flavors.soy [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+    <td>iphoneos-armv7</td>
+    <td>32-bit iPhone device</td>
+  </tr>
+  <tr>
+    <td>iphoneos-arm64</td>
+    <td>64-bit iPhone device</td>
+  </tr>
+  <tr>
+    <td>iphoneos-i386</td>
+    <td>WTF</td>
```
Comment (zpao) [inline docs/concept/flavors.soy:148]: Not appropriate. Please fix

File: docs/concept/flavors.soy [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+  </tr>
+  <tr>
+    <td>iphoneos-x86_64</td>
+    <td>WTF</td>
```
Comment (zpao) [inline docs/concept/flavors.soy:152]: same

File: docs/concept/flavors.soy [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+{/literal}
+
+<p>
+Symlink tree, or header map, of headers used to compile files for other 
+libraries that depend on this library.
+</p>
+
+<h2 id="apple_debug_http">Apple debug flavors<a class="inline-link" href="#apple_debug_http">#</a></h2>
+<p>
+Selects the actual binary used to represent a {call buck.cxx_binary /}. These are used internally.
```
Comment (zpao) [inline docs/concept/flavors.soy:374]: "used internally"? Internally to Buck or internal at FB?

File: docs/concept/flavors.soy [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+
+<h3>Other</h3>
+{literal} 
+<pre>
+apple-debuggable-binary
+</pre>
+{/literal}
+
+<p>
+[Description needed]
```
Comment (zpao) [inline docs/concept/flavors.soy:418]: Want to write it?

File: docs/concept/flavors.soy (hunk 4 above)
Comment (v-jizhang) [reply docs/concept/flavors.soy:374]: Buck

File: docs/concept/flavors.soy (hunk 5 above)
Comment (v-jizhang) [reply docs/concept/flavors.soy:418]: Done.

**Pull request merged**